### DISCUSSION TITLE
Source Date Support

### DISCUSSION THREAD

**Discussion by a4ginatl** [codefori/vscode-ibmi#2313] [Q&A]
I enabled this as it is useful and needed.

I get this message when starting up.

![image](https://github.com/user-attachments/assets/de6b94ff-756e-4899-b763-a7d885666705)

Why do I get this?
If its a provided option, why is it disabled?

Thanks
Darryl Freinkel.

<hr />

⚠️ **REMOVE THIS LINE AND ANY SENSITIVE INFORMATION BELOW!** ⚠️

|Context|Version|
|-|-|
|Code for IBM i version|2.13.5|
|Visual Studio Code version|1.94.2|
|Operating System|win32_x64|

<details>
<summary>Active extensions</summary>

```
COBOL (cobol): 24.10.26
Code for IBM i Walkthroughs (vscode-ibmi-walkthroughs): 0.5.0
Configuration Editing (configuration-editing): 1.0.0
Db2 for IBM i (vscode-db2i): 1.6.1
Emmet (emmet): 1.0.0
Error Lens (errorlens): 3.20.0
Git (git): 1.0.0
Git Base (git-base): 1.0.0
GitHub (github): 0.0.1
IBM i Debug (ibmidebug): 2.0.1
JSON Language Features (json-language-features): 1.0.0
Merge Conflict (merge-conflict): 1.0.0
Node Debug Auto-attach (debug-auto-launch): 1.0.0
RPGLE (vscode-rpgle): 0.26.12
TODO Highlight (vscode-todo-highlight): 1.0.5
TypeScript and JavaScript Language Features (typescript-language-features): 1.0.0
```
</details>

<hr />

<details>
<summary>Remote system</summary>

|Setting|Value|
|-|-|
|IBM i OS|V7R4M0|
|Tech Refresh|9|
|CCSID Origin|65535|
|Runtime CCSID|65535|
|Default CCSID|37|
|SQL|Enabled
|Source dates|Enabled

### Enabled features

|/QOpenSys/pkgs/bin|/usr/bin|/QSYS.lib/ILEDITOR.lib|/QSYS.LIB|/QIBM/ProdData/IBMiDebugService/bin|/QOpenSys/QIBM/ProdData/JavaVM/jdk80|/QOpenSys/QIBM/ProdData/JavaVM/jdk11|
|-|-|-|-|-|-|-|
|bash|attr|GETNEWLIBL.PGM|QZDFMDB2.PGM|startDebugService.sh|64bit|64bit|
|chsh|iconv||||||
|grep|setccsid||||||
|ls|tar||||||
|md5sum|||||||
|sort|||||||
|stat|||||||
|tn5250|||||||
</details>

<details>
<summary>Shell env</summary>

```bash
BUILDLIB=QGPL
CURLIB=QGPL
HOME=/home/dfreinkel
HOST=172.17.72.21
LIBLS=DPSRCLIB DPPDATA QGPL QTEMP OPRUTLLIB DFREINKEL
LOGIN=dfreinkel
LOGNAME=dfreinkel
MAIL=/var/spool/mail/dfreinkel
OLDPWD=/home/DFREINKEL
PASE_USRGRP_LIMITED=N
PATH=/QOpenSys/pkgs/bin:/QOpenSys/usr/bin:/usr/ccs/bin:/QOpenSys/usr/bin/X11:/usr/sbin:.:/usr/bin
PWD=/home/dfreinkel
SHELL=/QOpenSys/pkgs/bin/bash
SHLVL=1
SSH_CLIENT=10.27.230.147 51072 22
SSH_CONNECTION=10.27.230.147 51072 172.17.72.21 22
TZ=<EST>5<EDT>,M3.2.0,M11.1.0
USER=dfreinkel
USERNAME=DFREINKEL
WORKDIR=/home/dfreinkel
_=/QOpenSys/pkgs/bin/env
```
</details>

<details>
<summary>Variants</summary>

```json
{
  "american": "#@$",
  "local": "#@$"
}
```
</details>

<details>
<summary>Errors</summary>

```json
[
  {
    "command": "/QOpenSys/usr/bin/qsh",
    "code": 1,
    "stderr": "CPF2111:  Library ILEDITOR already exists.\nCPC2206:  Ownership of object QZSHSYSTEM in QTEMP type *USRSPC changed.",
    "cwd": "/home/dfreinkel"
  },
  {
    "command": "/QOpenSys/usr/bin/qsh",
    "code": 1,
    "stderr": "CPF2125:  No objects deleted.\nCPC2206:  Ownership of object QZSHSYSTEM in QTEMP type *USRSPC changed.",
    "cwd": "/home/dfreinkel"
  }
]
```
</details>

**Comment by worksofliam** [comment]
Looks like it is because your system QCCSID is 65535, in which we cannot enabled our improved SQL support.

**Reply by a4ginatl** [reply]
Why don't you use the CCSID on the user profile?
